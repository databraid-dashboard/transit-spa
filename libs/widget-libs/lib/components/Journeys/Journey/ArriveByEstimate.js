var _jsxFileName = '../../src/components/Journeys/Journey/ArriveByEstimate.jsx',
    _this = this;

import React from 'react';
import PropTypes from 'prop-types';

import { Icon, List } from 'semantic-ui-react';

var ArriveByEstimate = function ArriveByEstimate(_ref) {
  var eta = _ref.eta;
  return React.createElement(
    List.Item,
    {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 7
      },
      __self: _this
    },
    React.createElement(Icon, { name: 'clock', size: 'large', __source: {
        fileName: _jsxFileName,
        lineNumber: 8
      },
      __self: _this
    }),
    React.createElement(
      List.Content,
      {
        __source: {
          fileName: _jsxFileName,
          lineNumber: 9
        },
        __self: _this
      },
      React.createElement(
        List.Header,
        {
          __source: {
            fileName: _jsxFileName,
            lineNumber: 10
          },
          __self: _this
        },
        'ETA'
      ),
      eta
    )
  );
};

ArriveByEstimate.propTypes = {
  eta: PropTypes.string.isRequired
};

ArriveByEstimate.defaultProps = {
  eta: ''
};

export default ArriveByEstimate;