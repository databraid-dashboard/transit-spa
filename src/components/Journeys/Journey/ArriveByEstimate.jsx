import React from 'react';
import PropTypes from 'prop-types';

import { Icon, List } from 'semantic-ui-react';

const ArriveByEstimate = ({ eta }) =>
  (<List.Item>
    <Icon name="clock" size="large" />
    <List.Content>
      <List.Header>ETA</List.Header>
      {eta}
    </List.Content>
  </List.Item>);

ArriveByEstimate.propTypes = {
  eta: PropTypes.string.isRequired,
};

ArriveByEstimate.defaultProps = {
  eta: '',
};

export default ArriveByEstimate;
