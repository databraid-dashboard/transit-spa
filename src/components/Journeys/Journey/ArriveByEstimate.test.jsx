import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ArriveByEstimate from './ArriveByEstimate';

describe('ArriveByEstimate', () => {
  it('renders when given an eta', () => {
    const component = shallow(<ArriveByEstimate eta={'12:34pm'} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
