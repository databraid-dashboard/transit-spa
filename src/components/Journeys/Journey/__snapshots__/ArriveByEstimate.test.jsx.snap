// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArriveByEstimate renders when given an eta 1`] = `
<Header
  as="h4"
  style={
    Object {
      "marginBottom": "0rem",
      "marginTop": "0rem",
    }
  }
>
  12:34pm
</Header>
`;
